### ISSUE TITLE
Support for attributes

### ISSUE THREAD

**Issue by reckart** [pubannotation/pubannotation#8]
Currently, PubAnnotation can only store a single payload string per annotation in the `obj` property of the denotation or relation. Since most other annotation data models support key/value pairs to represent attributes of annotations, it would be good if PubAnnotation would also support these.

During BLAH5, a format along these lines was suggested:

```
attributes: [
  {"target": <annotation-id>, "key": <attribute-name>, "value": <attribute-value>}
]
```

* The target must be the ID of a denotation or relation annotation.
* The key must be a String value and must not contain whitespace.
* The attribute values are limited to primitive JSON-supported types (String, boolean, integer, float), i.e. no arrays or JSON objects.

**Comment by reckart** [comment]
@ksuderman @jdkim Tried to summarize here (and go into a slightly more concrete proposal).

**Comment by jdkim** [comment]
During the meeting last week, I could not exactly recall the format.
Now I checked the format, and its correct format is as follows:
"attributions":[
     {"subj":<string:denotation-id>, "pred":<string:attribute-name>,"obj":<string:attribute-value>}
]

**Comment by reckart** [comment]
Ok. Sure you want to stick with `attributions` instead of `attributes`?

**Comment by reckart** [comment]
I assume the`subj` can also be a relation-id, right?

**Comment by jdkim** [comment]
That is correct. Just it is not yet implemented in TextAE.
Here, you can find a version of TextAE which supports "attribution" type annotation:
http://textae-test.pubannotation.org/editor.html

**Comment by ksuderman** [comment]
I agree with Richard re: attributes vs attributions. If it is not too late I suggest using `attributes`.

**Comment by jdkim** [comment]
@reckart @ksuderman If "attributes" makes more sense to you, I do not have particular reason to stick to "attributions". Our goal is to have a easy-to-understand-and-remember names.
